### ISSUE TITLE
Add deprecation warning to Astro.resolve when used with the static build flag

### ISSUE THREAD

**Issue by matthewp** [withastro/astro#2315]

**Comment by FredKSchott** [comment]
- should not work behind flag
- deprecation warning should be without flag? probably too early

**Comment by matthewp** [comment]
CSS:

```
03:20 PM [deprecation] Astro.resolve() is deprecated. We see that you are trying to resolve ./styles/foo.css.
It looks like you are resolving styles. If you are adding a link tag, replace with this:

<style global>
@import "./styles/foo.css";
</style>
```

Other:

```
03:22 PM [deprecation] Astro.resolve() is deprecated. We see that you are trying to resolve ./images/penguin.png.
This can be replaced with a dynamic import like so: await import("./images/penguin.png")
```

**Comment by matthewp** [comment]
Closed by #2416
